### PR TITLE
fix: sequence classification metric and training bugs #780

### PR DESCRIPTION
**Pull request by yuhezhang-ai**
## PR Summary

The initial intent is to fix https://github.com/NVIDIA-NeMo/Automodel/issues/780 sequence classification metrics by adding proper accuracy, grad_norm, and TPS calculations, while removing irrelevant metrics - which I did, results now:
```
025-11-20 02:46:15 | INFO | root | step 0 | epoch 0 | loss 0.6957 | accuracy 0.5312 | grad_norm 152.2742 | lr 1.00e-07 | mem 4.63 GiB | tps 2624.70(2624.70/gpu)
2025-11-20 02:46:16 | INFO | root | step 1 | epoch 0 | loss 0.7005 | accuracy 0.5625 | grad_norm 19.4315 | lr 1.00e-07 | mem 4.54 GiB | tps 7309.84(7309.84/gpu)
2025-11-20 02:46:16 | INFO | root | step 2 | epoch 0 | loss 0.6744 | accuracy 0.5000 | grad_norm 14.5611 | lr 1.00e-07 | mem 4.24 GiB | tps 8090.28(8090.28/gpu)
2025-11-20 02:46:16 | INFO | root | step 3 | epoch 0 | loss 0.6654 | accuracy 0.7188 | grad_norm 9.8913 | lr 1.00e-07 | mem 4.12 GiB | tps 7819.38(7819.38/gpu)
```

But I found other issues while testing:

### 1. Base Model Loading Failure
**Problem**: Loading base RoBERTa for fine-tuning failed with `Missing key in checkpoint state_dict: classifier.dense.bias` because base models don't have the classifier heads. I got this error by running `python examples/llm_seq_cls/seq_cls.py -c examples/llm_seq_cls/glue/mrpc_roberta_lora.yaml`, and can only run successfully with `--checkpoint.load_base_model false`

**Fix**: I added `skip_task_head_prefixes_for_base_model` config parameter to filter out task head parameters when loading base models. Let me know if there is any issue or you want to fix in another way?

### 2. The Model is Not Training Correctly

The validation accuracy is constant after I trained for 15 epochs, and the predicted label is almost always 1 during training and validation.

**One Problem I Identified**: With LoRA enabled, the classifier head was frozen (only LoRA adapters were trainable), causing model to predict constant class. So I modified `build_model_and_optimizer` to accept `unfreeze_modules` parameter, allowing specific modules (like classifier heads) to be fully trainable alongside LoRA.

However, after the fix, the model is still not learning, and I also tried different hyperparams like learning rate etc. I wonder is there any deeper issue, or I misunderstand something?

### CODE REVIEW

**Comment by yuhezhang-ai** [comment]
> @yuhezhang-ai Could you run formatting on your changes by running `uv run ruff format .` in your branch? Thanks

Done! I've run ruff format and also rebased to sign off all commits (forgot to sign off previously...)

**Quick note on training investigation:**
I initially observed validation accuracy stuck at the majority baseline (~68%) during testing. I thought it's an internal bug. But after debugging with a vanilla PyTorch baseline (which worked fine), I discovered it was a hyperparameter issue, not a bug in the repo.

**Updated the example config (mrpc_roberta_lora.yaml) with stable hyperparameters:**
lr: 3e-4 → 2e-5 (standard for BERT/RoBERTa)
weight_decay: 0 → 0.01 (crucial for stable training on small datasets)

Training now converges properly. The val accuracy is about 87% after 15 epochs! Logs:

```
2025-11-20 17:26:06 | INFO | root | step 1696 | epoch 14 | loss 0.1132 | accuracy 1.0000 | grad_norm 24.9916 | lr 2.00e-05 | mem 4.21 GiB | tps 4152.42(4152.42/gpu)
2025-11-20 17:26:06 | INFO | root | step 1697 | epoch 14 | loss 0.3781 | accuracy 0.8438 | grad_norm 73.6510 | lr 2.00e-05 | mem 4.08 GiB | tps 4142.08(4142.08/gpu)
2025-11-20 17:26:07 | INFO | root | step 1698 | epoch 14 | loss 0.0525 | accuracy 1.0000 | grad_norm 16.3862 | lr 2.00e-05 | mem 4.70 GiB | tps 3481.89(3481.89/gpu)
2025-11-20 17:26:07 | INFO | root | step 1699 | epoch 14 | loss 0.1471 | accuracy 0.9688 | grad_norm 35.0635 | lr 2.00e-05 | mem 4.35 GiB | tps 3868.34(3868.34/gpu)
2025-11-20 17:26:10 | INFO | root | [val] step 1699 | epoch 14 | loss 0.3292 | accuracy 0.8725 | lr 2.00e-05
```

**Review by akoumpa — APPROVED** [review]

**Comment by akoumpa** [comment]
Hi @NVIDIA-NeMo/automation , can you FM this? i've edited the codecov.yaml to be based on overall score rather than patch, current code has previously passed https://github.com/NVIDIA-NeMo/Automodel/actions/runs/19546147884/job/55979344052

**Review by pablo-garay — APPROVED** [review]
